### PR TITLE
feat: [#188283234] Enable contextual info in non essential questions

### PR DESCRIPTION
**Pull request by SirSamuelJoseph**
## Description
Enables embedding contextual links within non-essential questions

### Ticket

This pull request resolves [#188283234](https://www.pivotaltracker.com/story/show/188283234).

### Approach
Converts non-essential question to Content.

This format `text|contextual-info-text` will be displayed in the CMS as a string, not as a contextual text. Right now this only affects a single NEQ but we might need a follow-up ticket so it properly displays in the CMS

### Steps to Test

1. Create new Poppy in health care industry
2. Go to profile to see non-essential question

## Code author checklist

- [x] I have rebased this branch from the latest main branch
- [x] I have performed a self-review of my code
- [x] I have created and/or updated relevant documentation on the engineering documentation website
- [x] I have not used any relative imports
- [x] I have pruned any instances of unused code
- [x] I have not added any markdown to labels, titles and button text in config
- [x] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [x] I have checked for and removed instances of unused config from CMS
- [x] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts` (see CMS Additions in Engineering Reference/FAQ on the engineering documentation site)
- [x] I have updated relevant `.env` values in both `.env-template` and in Bitwarden

### CODE REVIEW

**Comment by jphechter** [comment]
I don't have a problem with this pattern. Storing strings that will be formatted, like this contextual link, in the JSON files isn't a problem. The only issue I could see is this being rendered in the CMS as just plain text.

We could change this widget from `string` to `markdown`. That will at least give the content team a greater understanding of what the string represents.
Screenshot 2024-09-16 at 1 27 03 PM

It should also help with any future edits they need to make.
Screenshot 2024-09-16 at 1 27 36 PM

This might also be a good candidate for a CMS preview on the right-hand side. I think we should be able to do something programmatic, like looping over all the objects in `nonEssentialQuestions.json`, so that this could be reflective of any new questions they add.

TL;DR version
No issues with the pattern, but I think there's more we can do to make this more helpful for the CMS users.
@SirSamuelJoseph @seidior

**Review by somabadri — APPROVED** [review]
+1 to what jp said but otherwise it looks good to me!

File: web/src/components/data-fields/non-essential-questions/NonEssentialQuestion.tsx [hunk 1]
```diff
@@ -28,7 +29,9 @@ export const NonEssentialQuestion = (props: Props): ReactElement => {
       {nonEssentialQuestionText && (
         <>
           <div className={"margin-top-2"}>
-            <span className={"text-bold"}>{nonEssentialQuestionText}</span>
+            <span className={"text-bold"}>
+              <Content>{nonEssentialQuestionText}</Content>
+            </span>
```
Comment (seidior) [inline web/src/components/data-fields/non-essential-questions/NonEssentialQuestion.tsx:34]: [sand] I know it's mostly meaningless these days, but since `<Content>` can include block-level elements, it should be wrapped in a block-level element rather than a `<span>`.

**Review by seidior — COMMENTED** [review]
Thanks, Sam!
